Editor: Display warning when post type not supported by site

aduth opened the pull request:
Related: #5398 

This pull request seeks to display a warning in the post editor when attempting to edit a post type which is not supported by the site.

Before|After
---|---
![Before](https://cloud.githubusercontent.com/assets/1779930/15563321/8475a6f2-22d5-11e6-8a4b-dd06ceea9f4a.png)|![After](https://cloud.githubusercontent.com/assets/1779930/15830586/30a9ea36-2be7-11e6-9115-834bcc4b406f.png)

__Testing instructions:__

Ensure there are no regressions or warnings when attempting to edit an existing post type, particularly posts or pages.

Verify that when attempting to edit a post type which does not exist for the site, the warning is shown, e.g. [`/edit/xyz`](http://calypso.localhost:3000/edit/xyz) or [`/edit/jetpack-portfolio`](http://calypso.localhost:3000/edit/xyz) when portfolios are not enabled.

folletto commented:
Tested on Chrome, seems working well. :)

A couple of notes:

1. What's the reason to redirect to `stats`? Wouldn't be better to redirect on Posts? I'm saying this mostly guessing the user intent.
2. Can we swap the order? Blue-right (primary), white-left?
3. Could we write a text that gives a better idea of what's happening? If I was a user and I get this message, I get the feeling of an error, but I would have no clue about what's happening. Can we maybe swap the copy / buttons to have a link to a support page that tells what CPTs are?

Quick draft to show with an example what I mean:

> #### This post type is not supported
> To use this post type, please check that your site supports it.
> To read more about this, check our support page on custom post types.
> 
> [ Return to My Sites ] [ Write a normal post ]

Note: I'm not entirely sure what are the requirements to _"support"_ a CPT, so the message above is probably a bit off and could be even more clear. :)

aduth commented:
Thanks for the feedback @folletto ! Admittedly I made a few barely-founded assumptions here, so happy to make large revisions.

The scenario here is not a common one, but perhaps a user was referred to edit testimonials by the link `/edit/jetpack-testimonials`. They choose a site, but it hasn't enabled the post type. Currently, the editor loads and gives the user the false impression they can start writing. This isn't exclusive to our own supported post types, e.g. if a user manually edits the path with a nonsensical post type.

Related: #5329 (list view equivalent)

>What's the reason to redirect to `stats`? Wouldn't be better to redirect on Posts? I'm saying this mostly guessing the user intent.

Yeah, I thought it useful to include some sort of "Get me out of here!" link, and stats (the default landing for My Sites) came to mind, though I do think Posts may be more useful for what a user would be trying to achieve if caught in this scenario.

>Can we swap the order? Blue-right (primary), white-left?

Sure.

>Could we write a text that gives a better idea of what's happening?

Yep, in retrospect, the text isn't very helpful or comforting. Also consider that, while related, this doesn't yet satisfy the requirements of #5398, which is: if a site supports a particular post type, but the user doesn't meet the requirements to be able to manage that type of post. In those cases, we'll likely want a similar experience (in case we can think of text that applies well to both).

What you've suggested seems to work well, though as mentioned above, this isn't exclusive to post types that we offer on WordPress.com/Jetpack. I'd suggest we'd either change the "read more" line of your proposal to something more of a suggestion ("Check out our custom post types!") or conditionally shown only if the attempted type trying to be accessed matches one of our supported types (as we did in #5329).

I'm also not sure "normal post" conveys any more meaning. Maybe "blog post" ? Maybe eliminate this button altogether?

aduth commented:
Hmm... having trouble finding it, but didn't we have an option for `<Dialog />` to not obscure the master bar? That might be a good option to preventing the user from feeling imprisoned in the editor. That said, we already have a couple other dialogs in the editor which are not using this (e.g. post trashed warning).

folletto commented:
Thanks @aduth for the quick chat to clarify some details behind the activation logic of CPTs, and the iteration on copy. 

We reached i3, where we differentiate text between:

* CPTs part of WordPress.com
* CPTs that we don't know about
* User without edit rights

![calypso-cpt-not-supported-i3](https://cloud.githubusercontent.com/assets/4389/15611854/2ef6acda-2424-11e6-9d87-f9c88f6041d8.png)

Note that this will require a new page in the support docs that explains briefly Custom Post Types. Following up on that separately. :)

folletto commented:
Ok, support page created: https://en.support.wordpress.com/custom-post-types/

aduth commented:
Rebased, implemented updated wording (and consideration for "configurable types") and updated screenshots/testing instructions from original comment accordingly.

Note "No CPT user access" concept is not implemented here and will be added in a subsequent pull request.

folletto commented:
Looks good! 👍

timmyc commented:
Working well for me, after a rebase this should be good to 🚢

aduth commented on client/post-editor/editor-post-type-unsupported/index.jsx:
```diff
@@ -0,0 +1,118 @@
+/**
+ * Internal dependencies
+ */
+import localize from 'lib/mixins/i18n/localize';
```
>Hi there, i18n usage has changed recently.

@Tug Thanks for the heads-up, updated in the rebased ce29590.
